plugins/openpgp-api-lib/openpgp-api/src/main/java/org/openintents/openpgp/AutocryptPeerUpdate.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2014-2015 Dominik Schürmann <dominik@dominikschuermann.de>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openintents.openpgp;
+
+
+import java.util.Date;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+
+@SuppressWarnings("unused")
+public class AutocryptPeerUpdate implements Parcelable {
+    /**
+     * Since there might be a case where new versions of the client using the library getting
+     * old versions of the protocol (and thus old versions of this class), we need a versioning
+     * system for the parcels sent between the clients and the providers.
+     */
+    private static final int PARCELABLE_VERSION = 1;
+
+
+    private final byte[] keyData;
+    private final Date effectiveDate;
+    private final PreferEncrypt preferEncrypt;
+
+
+    private AutocryptPeerUpdate(byte[] keyData, Date effectiveDate, PreferEncrypt preferEncrypt) {
+        this.keyData = keyData;
+        this.effectiveDate = effectiveDate;
+        this.preferEncrypt = preferEncrypt;
+    }
+
+    private AutocryptPeerUpdate(Parcel source, int version) {
+        this.keyData = source.createByteArray();
+        this.effectiveDate = source.readInt() != 0 ? new Date(source.readLong()) : null;
+        this.preferEncrypt = PreferEncrypt.values()[source.readInt()];
+    }
+
+
+    public static AutocryptPeerUpdate createAutocryptPeerUpdate(byte[] keyData, Date timestamp, boolean isMutual) {
```

Comment:
`AutocryptPeerUpdate.createAutocryptPeerUpdate(…)` seems very redundant. Maybe just call the method `create`?